code/modules/spells/spell_types/disguise.dm

```diff
@@ -0,0 +1,71 @@
+	desc = "Why fight your foes when you can simply outwit them? Disguises the caster as a random crewmember."
+	invocation = "CONJR DIS GUISE"
+	invocation_type = "whisper"
+	school = "transmutation"
+	charge_max = 30 SECONDS
+	level_max = 2
+	cooldown_min = 25 SECONDS
+	clothes_req = FALSE
+	var/is_disguised = FALSE //Tells us if a disguise is currently up.
+	action_icon = 'icons/mob/actions/actions_spells.dmi'
+	action_icon_state = "disguise"
+
+/obj/effect/proc_holder/spell/disguise/can_cast(mob/user = usr)
+	if(!ishuman(user))
+		to_chat(user, span_danger("You cannot disguise as a non-humanoid!")) //The mob is not a human, so they can't disguise.
+		//We need to undo the cloak after non-humanoid disguises because when the wizard becomes a non human during the spell, it will mess up their sprite. But since they are non human, we can't actually undo the spell. This leaves our recloaking bugged as hell, and breaks a lot of stuff.
+		return FALSE
+	return TRUE
+
+/obj/effect/proc_holder/spell/disguise/choose_targets(mob/user = usr)
+	perform(user=user)
+
+/obj/effect/proc_holder/spell/disguise/cast(list/targets, mob/user = usr)
+	var/mob/living/carbon/human/C = user //Turns the user into a carbon, we'll need this later.
+	var/list/potentials = list()
+	for(var/mob/living/carbon/human/H in GLOB.carbon_list) //Checks all the humanoid mobs.
+		if(H.job) //Checks if they're crew.
+			potentials += H //Adds the crewmember to the list.
+	if(potentials.len == 0)
+		to_chat(C, span_notice("There's nobody to disguise as!"))
+		revert_cast(user)
+		return
+	var/mob/living/carbon/human/target = pick(potentials) //Picks a random subject from the viable targets.
+	cloak(C, target)
+
+/obj/effect/proc_holder/spell/disguise/proc/cloak(var/mob/living/carbon/human/C, var/mob/living/carbon/human/target) //Code shortcut to enable the disguise.
+	if(is_disguised)
+		message_admins("[ADMIN_LOOKUPFLW(C)] just tried to disguise while disguised! That shouldn't happen!") 
+		return
+	C.name_override = target.name
+	C.SetSpecialVoice(target.name)
+	new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(C), C.dir) //Disguise animation.
+	C.icon = target.icon
+	C.icon_state = target.icon_state
+	C.cut_overlays()
+	C.add_overlay(target.get_overlays_copy(list(HANDS_LAYER)))
+	C.update_inv_hands()
+	log_game("[C.name] has disguised as [target.name]!") 
+	is_disguised = TRUE
+	addtimer(CALLBACK(src, .proc/undocloak, C), (15 SECONDS + (spell_level * 3 SECONDS)))
+	return
+
+/obj/effect/proc_holder/spell/disguise/proc/undocloak(var/mob/living/carbon/human/C) //Code shortcut to disable the disguise.
+	if(!is_disguised)
+		log_game("[C.name] just undid their disguise while not disguised, shouldn't cause harm but it's weird.") 
+		return
+	new /obj/effect/temp_visual/dir_setting/ninja(get_turf(C), C.dir) //Makes an animation for disguising.
+	C.name_override = null
+	C.UnsetSpecialVoice()
+	C.cut_overlays()
+	C.regenerate_icons()
+	is_disguised = FALSE
+	return

```

Comment:
```suggestion
/obj/effect/proc_holder/spell/disguise
	name = "Mimicry"
	desc = "Why fight your foes when you can simply outwit them? Disguises the caster as a random crewmember."
	invocation = "CONJR DIS GUISE"
	invocation_type = "whisper"
	school = "transmutation"
	charge_max = 30 SECONDS
	level_max = 2
	cooldown_min = 25 SECONDS
	clothes_req = FALSE
	var/is_disguised = FALSE //Tells us if a disguise is currently up.
	action_icon = 'icons/mob/actions/actions_spells.dmi'
	action_icon_state = "disguise"

/obj/effect/proc_holder/spell/disguise/can_cast(mob/user = usr)
	if(!ishuman(user))
		to_chat(user, span_danger("You cannot disguise as a non-humanoid!")) //The mob is not a human, so they can't disguise.
		//We need to undo the cloak after non-humanoid disguises because when the wizard becomes a non human during the spell, it will mess up their sprite. But since they are non human, we can't actually undo the spell. This leaves our recloaking bugged as hell, and breaks a lot of stuff.
		return FALSE
	return TRUE

/obj/effect/proc_holder/spell/disguise/choose_targets(mob/user = usr)
	perform(user=user)

/obj/effect/proc_holder/spell/disguise/cast(list/targets, mob/living/carbon/C = usr) //Turns the user into a carbon, we'll need this later.
	var/list/potentials = list()
	for(var/mob/living/carbon/human/H in GLOB.carbon_list) //Checks all the humanoid mobs.
		if(H.job) //Checks if they're crew.
			potentials += H //Adds the crewmember to the list.
	if(!potentials.len)
		to_chat(C, span_notice("There's nobody to disguise as!"))
		revert_cast(user)
		return
	var/mob/living/carbon/human/target = pick(potentials) //Picks a random subject from the viable targets.
	cloak(C, target)

/obj/effect/proc_holder/spell/disguise/proc/cloak(mob/living/L, mob/living/target) //Code shortcut to enable the disguise.
	if(is_disguised)
		message_admins("[ADMIN_LOOKUPFLW(C)] just tried to disguise while disguised! That shouldn't happen!") 
		return
	new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(L), L.dir) //Disguise animation.
	if(ishuman(L))
		var/mob/living/carbon/human/H = L
		H.name_override = target.name
		H.SetSpecialVoice(target.name)
		H.icon = target.icon 
		H.icon_state = target.icon_state
		H.update_inv_hands()
	L.cut_overlays()
	L.add_overlay(target.get_overlays_copy(list(HANDS_LAYER)))
	log_game("[L.name] has disguised as [target.name]!") 
	is_disguised = TRUE
	addtimer(CALLBACK(src, .proc/undocloak, L), (15 SECONDS + (spell_level * 3 SECONDS)))

/obj/effect/proc_holder/spell/disguise/proc/undocloak(mob/living/L) //Code shortcut to disable the disguise.
	if(!is_disguised)
		log_game("[L.name] just undid their disguise while not disguised, shouldn't cause harm but it's weird.") 
		return
	new /obj/effect/temp_visual/dir_setting/ninja(get_turf(L), L.dir) //Makes an animation for disguising.
	if(ishuman(L))
		var/mob/living/carbon/human/H = L
		H.name_override = null
		H.UnsetSpecialVoice()
	L.cut_overlays()
	L.regenerate_icons()
	is_disguised = FALSE

```

Reply:
problem probably is if you shapeshift it runtimes when it runtimes the uncloak proc, see the dream deamon panel or the runtime checker in the admin tabs for these

Reply:
This wouldn't fix it, because if they were human during the undisguise it would just leave their name messed up.

Reply:
Rather, if they are NON human

Reply:
oh wait you have a genius idea, nevermind LOL